tests/e2e/scenarios/test_provider_fault_proxy.py

```
@@ -146,6 +149,113 @@ async def test_response_fault_profiles_do_not_reach_provider(
+        "untested": sorted(set(PROVIDER_FAULT_PROFILES) - covered),
+        "stale": sorted(covered - set(PROVIDER_FAULT_PROFILES)),
+    }
+
+
+@pytest.mark.parametrize(
+    ("profile_name", "status", "challenge_contains"),
+    (
+        ("missing_credential", 401, 'Bearer realm='),
+        ("expired_credential", 401, 'error="invalid_token"'),
+        ("wrong_scope", 403, 'error="insufficient_scope"'),
+    ),
```

Ruff (0.15.21)

[warning] Wrong values type in `pytest.mark.parametrize` expected `list` of `tuple`

Use `list` of `tuple` for parameter values

(PT007)